doc/lua_api.txt

```diff
@@ -1118,54 +1118,109 @@ A box of a regular node would look like:
 
     {-0.5, -0.5, -0.5, 0.5, 0.5, 0.5},
 
+Perlin noise
+------------
+Perlin noise creates a continuously-varying value depending on the input values.
+Usually in Minetest the input values are either 2D or 3D co-ordinates in nodes.
+The result is mostly used during mapgen to create the terrain shape, or to vary
+the density of decorations or vary the structure of ores.
+
+### Structure of perlin noise
+An 'octave' is a simple noise generator that outputs a value between -1 and 1.
+The smooth wavy noise it generates has a single characteristic scale, almost
+like a 'wavelength', so on its own does not create fine detail.
```

Comment:
Smoothness results from interpolating of low frequency noise and high frequency noises with a reduced strength. Maybe we should prefer to keep it simple and refer to a other site to explain how perlin noise works, so we can use that to explain the API easily.

Reply:
I want to explain here because i think i can explain well and searching for a clear explanation can be difficult.
This is incorrect :) :
`Smoothness results from interpolating of low frequency noise and high frequency noises with a reduced strength.`

Reply:
I don't think we should link to external sites for this. The explanation can be here, as long as it's kept brief and simple.